<!doctype html>
<html>
<head>
    <meta charset='UTF-8'>
    <meta name='viewport' content='width=device-width initial-scale=1'>
    <meta name="description"; content="single-channel multi-unit spiking model">
    <meta name="keywords"; content="RC, Rat, Reaching, Rat Reaching, Rehabilitation, Stroke, Preclinical Studies, Neurophysiology, Motor Cortex, Premotor Cortex">
    <meta name="author"; content="Max Murphy">
<title>RC-Project</title><link href='https://fonts.loli.net/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext' rel='stylesheet' type='text/css' /><style type='text/css'>html {overflow-x: initial !important;}:root { --bg-color:#ffffff; --text-color:#333333; --select-text-bg-color:#B5D6FC; --select-text-font-color:auto; --monospace:"Lucida Console",Consolas,"Courier",monospace; }
html { font-size: 14px; background-color: var(--bg-color); color: var(--text-color); font-family: "Helvetica Neue", Helvetica, Arial, sans-serif; -webkit-font-smoothing: antialiased; }
body { margin: 0px; padding: 0px; height: auto; bottom: 0px; top: 0px; left: 0px; right: 0px; font-size: 1rem; line-height: 1.42857; overflow-x: hidden; background: inherit; tab-size: 4; }
iframe { margin: auto; }
a.url { word-break: break-all; }
a:active, a:hover { outline: 0px; }
.in-text-selection, ::selection { text-shadow: none; background: var(--select-text-bg-color); color: var(--select-text-font-color); }
#write { margin: 0px auto; height: auto; width: inherit; word-break: normal; overflow-wrap: break-word; position: relative; white-space: normal; overflow-x: visible; padding-top: 40px; }
#write.first-line-indent p { text-indent: 2em; }
#write.first-line-indent li p, #write.first-line-indent p * { text-indent: 0px; }
#write.first-line-indent li { margin-left: 2em; }
.for-image #write { padding-left: 8px; padding-right: 8px; }
body.typora-export { padding-left: 30px; padding-right: 30px; }
.typora-export .footnote-line, .typora-export li, .typora-export p { white-space: pre-wrap; }
@media screen and (max-width: 500px) {
  body.typora-export { padding-left: 0px; padding-right: 0px; }
  #write { padding-left: 20px; padding-right: 20px; }
  .CodeMirror-sizer { margin-left: 0px !important; }
  .CodeMirror-gutters { display: none !important; }
}
#write li > figure:last-child { margin-bottom: 0.5rem; }
#write ol, #write ul { position: relative; }
img { max-width: 100%; vertical-align: middle; }
button, input, select, textarea { color: inherit; font: inherit; }
input[type="checkbox"], input[type="radio"] { line-height: normal; padding: 0px; }
*, ::after, ::before { box-sizing: border-box; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p, #write pre { width: inherit; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p { position: relative; }
p { line-height: inherit; }
h1, h2, h3, h4, h5, h6 { break-after: avoid-page; break-inside: avoid; orphans: 4; }
p { orphans: 4; }
h1 { font-size: 2rem; }
h2 { font-size: 1.8rem; }
h3 { font-size: 1.6rem; }
h4 { font-size: 1.4rem; }
h5 { font-size: 1.2rem; }
h6 { font-size: 1rem; }
.md-math-block, .md-rawblock, h1, h2, h3, h4, h5, h6, p { margin-top: 1rem; margin-bottom: 1rem; }
.hidden { display: none; }
.md-blockmeta { color: rgb(204, 204, 204); font-weight: 700; font-style: italic; }
a { cursor: pointer; }
sup.md-footnote { padding: 2px 4px; background-color: rgba(238, 238, 238, 0.7); color: rgb(85, 85, 85); border-radius: 4px; cursor: pointer; }
sup.md-footnote a, sup.md-footnote a:hover { color: inherit; text-transform: inherit; text-decoration: inherit; }
#write input[type="checkbox"] { cursor: pointer; width: inherit; height: inherit; }
figure { overflow-x: auto; margin: 1.2em 0px; max-width: calc(100% + 16px); padding: 0px; }
figure > table { margin: 0px !important; }
tr { break-inside: avoid; break-after: auto; }
thead { display: table-header-group; }
table { border-collapse: collapse; border-spacing: 0px; width: 100%; overflow: auto; break-inside: auto; text-align: left; }
table.md-table td { min-width: 32px; }
.CodeMirror-gutters { border-right: 0px; background-color: inherit; }
.CodeMirror-linenumber { user-select: none; }
.CodeMirror { text-align: left; }
.CodeMirror-placeholder { opacity: 0.3; }
.CodeMirror pre { padding: 0px 4px; }
.CodeMirror-lines { padding: 0px; }
div.hr:focus { cursor: none; }
#write pre { white-space: pre-wrap; }
#write.fences-no-line-wrapping pre { white-space: pre; }
#write pre.ty-contain-cm { white-space: normal; }
.CodeMirror-gutters { margin-right: 4px; }
.md-fences { font-size: 0.9rem; display: block; break-inside: avoid; text-align: left; overflow: visible; white-space: pre; background: inherit; position: relative !important; }
.md-diagram-panel { width: 100%; margin-top: 10px; text-align: center; padding-top: 0px; padding-bottom: 8px; overflow-x: auto; }
#write .md-fences.mock-cm { white-space: pre-wrap; }
.md-fences.md-fences-with-lineno { padding-left: 0px; }
#write.fences-no-line-wrapping .md-fences.mock-cm { white-space: pre; overflow-x: auto; }
.md-fences.mock-cm.md-fences-with-lineno { padding-left: 8px; }
.CodeMirror-line, twitterwidget { break-inside: avoid; }
.footnotes { opacity: 0.8; font-size: 0.9rem; margin-top: 1em; margin-bottom: 1em; }
.footnotes + .footnotes { margin-top: 0px; }
.md-reset { margin: 0px; padding: 0px; border: 0px; outline: 0px; vertical-align: top; background: 0px 0px; text-decoration: none; text-shadow: none; float: none; position: static; width: auto; height: auto; white-space: nowrap; cursor: inherit; -webkit-tap-highlight-color: transparent; line-height: normal; font-weight: 400; text-align: left; box-sizing: content-box; direction: ltr; }
li div { padding-top: 0px; }
blockquote { margin: 1rem 0px; }
li .mathjax-block, li p { margin: 0.5rem 0px; }
li { margin: 0px; position: relative; }
blockquote > :last-child { margin-bottom: 0px; }
blockquote > :first-child, li > :first-child { margin-top: 0px; }
.footnotes-area { color: rgb(136, 136, 136); margin-top: 0.714rem; padding-bottom: 0.143rem; white-space: normal; }
#write .footnote-line { white-space: pre-wrap; }
@media print {
  body, html { border: 1px solid transparent; height: 99%; break-after: avoid; break-before: avoid; font-variant-ligatures: no-common-ligatures; }
  #write { margin-top: 0px; padding-top: 0px; border-color: transparent !important; }
  .typora-export * { -webkit-print-color-adjust: exact; }
  html.blink-to-pdf { font-size: 13px; }
  .typora-export #write { padding-left: 32px; padding-right: 32px; padding-bottom: 0px; break-after: avoid; }
  .typora-export #write::after { height: 0px; }
}
.footnote-line { margin-top: 0.714em; font-size: 0.7em; }
a img, img a { cursor: pointer; }
pre.md-meta-block { font-size: 0.8rem; min-height: 0.8rem; white-space: pre-wrap; background: rgb(204, 204, 204); display: block; overflow-x: hidden; }
p > .md-image:only-child:not(.md-img-error) img, p > img:only-child { display: block; margin: auto; }
#write.first-line-indent p > .md-image:only-child:not(.md-img-error) img { left: -2em; position: relative; }
p > .md-image:only-child { display: inline-block; width: 100%; }
#write .MathJax_Display { margin: 0.8em 0px 0px; }
.md-math-block { width: 100%; }
.md-math-block:not(:empty)::after { display: none; }
[contenteditable="true"]:active, [contenteditable="true"]:focus, [contenteditable="false"]:active, [contenteditable="false"]:focus { outline: 0px; box-shadow: none; }
.md-task-list-item { position: relative; list-style-type: none; }
.task-list-item.md-task-list-item { padding-left: 0px; }
.md-task-list-item > input { position: absolute; top: 0px; left: 0px; margin-left: -1.2em; margin-top: calc(1em - 10px); border: none; }
.math { font-size: 1rem; }
.md-toc { min-height: 3.58rem; position: relative; font-size: 0.9rem; border-radius: 10px; }
.md-toc-content { position: relative; margin-left: 0px; }
.md-toc-content::after, .md-toc::after { display: none; }
.md-toc-item { display: block; color: rgb(65, 131, 196); }
.md-toc-item a { text-decoration: none; }
.md-toc-inner:hover { text-decoration: underline; }
.md-toc-inner { display: inline-block; cursor: pointer; }
.md-toc-h1 .md-toc-inner { margin-left: 0px; font-weight: 700; }
.md-toc-h2 .md-toc-inner { margin-left: 2em; }
.md-toc-h3 .md-toc-inner { margin-left: 4em; }
.md-toc-h4 .md-toc-inner { margin-left: 6em; }
.md-toc-h5 .md-toc-inner { margin-left: 8em; }
.md-toc-h6 .md-toc-inner { margin-left: 10em; }
@media screen and (max-width: 48em) {
  .md-toc-h3 .md-toc-inner { margin-left: 3.5em; }
  .md-toc-h4 .md-toc-inner { margin-left: 5em; }
  .md-toc-h5 .md-toc-inner { margin-left: 6.5em; }
  .md-toc-h6 .md-toc-inner { margin-left: 8em; }
}
a.md-toc-inner { font-size: inherit; font-style: inherit; font-weight: inherit; line-height: inherit; }
.footnote-line a:not(.reversefootnote) { color: inherit; }
.md-attr { display: none; }
.md-fn-count::after { content: "."; }
code, pre, samp, tt { font-family: var(--monospace); }
kbd { margin: 0px 0.1em; padding: 0.1em 0.6em; font-size: 0.8em; color: rgb(36, 39, 41); background: rgb(255, 255, 255); border: 1px solid rgb(173, 179, 185); border-radius: 3px; box-shadow: rgba(12, 13, 14, 0.2) 0px 1px 0px, rgb(255, 255, 255) 0px 0px 0px 2px inset; white-space: nowrap; vertical-align: middle; }
.md-comment { color: rgb(162, 127, 3); opacity: 0.8; font-family: var(--monospace); }
code { text-align: left; vertical-align: initial; }
a.md-print-anchor { white-space: pre !important; border-width: initial !important; border-style: none !important; border-color: initial !important; display: inline-block !important; position: absolute !important; width: 1px !important; right: 0px !important; outline: 0px !important; background: 0px 0px !important; text-decoration: initial !important; text-shadow: initial !important; }
.md-inline-math .MathJax_SVG .noError { display: none !important; }
.html-for-mac .inline-math-svg .MathJax_SVG { vertical-align: 0.2px; }
.md-math-block .MathJax_SVG_Display { text-align: center; margin: 0px; position: relative; text-indent: 0px; max-width: none; max-height: none; min-height: 0px; min-width: 100%; width: auto; overflow-y: hidden; display: block !important; }
.MathJax_SVG_Display, .md-inline-math .MathJax_SVG_Display { width: auto; margin: inherit; display: inline-block !important; }
.MathJax_SVG .MJX-monospace { font-family: var(--monospace); }
.MathJax_SVG .MJX-sans-serif { font-family: sans-serif; }
.MathJax_SVG { display: inline; font-style: normal; font-weight: 400; line-height: normal; zoom: 90%; text-indent: 0px; text-align: left; text-transform: none; letter-spacing: normal; word-spacing: normal; overflow-wrap: normal; white-space: nowrap; float: none; direction: ltr; max-width: none; max-height: none; min-width: 0px; min-height: 0px; border: 0px; padding: 0px; margin: 0px; }
.MathJax_SVG * { transition: none 0s ease 0s; }
.MathJax_SVG_Display svg { vertical-align: middle !important; margin-bottom: 0px !important; margin-top: 0px !important; }
.os-windows.monocolor-emoji .md-emoji { font-family: "Segoe UI Symbol", sans-serif; }
.md-diagram-panel > svg { max-width: 100%; }
[lang="flow"] svg, [lang="mermaid"] svg { max-width: 100%; height: auto; }
[lang="mermaid"] .node text { font-size: 1rem; }
table tr th { border-bottom: 0px; }
video { max-width: 100%; display: block; margin: 0px auto; }
iframe { max-width: 100%; width: 100%; border: none; }
.highlight td, .highlight tr { border: 0px; }
svg[id^="mermaidChart"] { line-height: 1em; }
mark { background: rgb(255, 255, 0); color: rgb(0, 0, 0); }
.md-html-inline .md-plain, .md-html-inline strong, mark .md-inline-math, mark strong { color: inherit; }
mark .md-meta { color: rgb(0, 0, 0); opacity: 0.3 !important; }


.CodeMirror { height: auto; }
.CodeMirror.cm-s-inner { background: inherit; }
.CodeMirror-scroll { overflow: auto hidden; z-index: 3; }
.CodeMirror-gutter-filler, .CodeMirror-scrollbar-filler { background-color: rgb(255, 255, 255); }
.CodeMirror-gutters { border-right: 1px solid rgb(221, 221, 221); background: inherit; white-space: nowrap; }
.CodeMirror-linenumber { padding: 0px 3px 0px 5px; text-align: right; color: rgb(153, 153, 153); }
.cm-s-inner .cm-keyword { color: rgb(119, 0, 136); }
.cm-s-inner .cm-atom, .cm-s-inner.cm-atom { color: rgb(34, 17, 153); }
.cm-s-inner .cm-number { color: rgb(17, 102, 68); }
.cm-s-inner .cm-def { color: rgb(0, 0, 255); }
.cm-s-inner .cm-variable { color: rgb(0, 0, 0); }
.cm-s-inner .cm-variable-2 { color: rgb(0, 85, 170); }
.cm-s-inner .cm-variable-3 { color: rgb(0, 136, 85); }
.cm-s-inner .cm-string { color: rgb(170, 17, 17); }
.cm-s-inner .cm-property { color: rgb(0, 0, 0); }
.cm-s-inner .cm-operator { color: rgb(152, 26, 26); }
.cm-s-inner .cm-comment, .cm-s-inner.cm-comment { color: rgb(170, 85, 0); }
.cm-s-inner .cm-string-2 { color: rgb(255, 85, 0); }
.cm-s-inner .cm-meta { color: rgb(85, 85, 85); }
.cm-s-inner .cm-qualifier { color: rgb(85, 85, 85); }
.cm-s-inner .cm-builtin { color: rgb(51, 0, 170); }
.cm-s-inner .cm-bracket { color: rgb(153, 153, 119); }
.cm-s-inner .cm-tag { color: rgb(17, 119, 0); }
.cm-s-inner .cm-attribute { color: rgb(0, 0, 204); }
.cm-s-inner .cm-header, .cm-s-inner.cm-header { color: rgb(0, 0, 255); }
.cm-s-inner .cm-quote, .cm-s-inner.cm-quote { color: rgb(0, 153, 0); }
.cm-s-inner .cm-hr, .cm-s-inner.cm-hr { color: rgb(153, 153, 153); }
.cm-s-inner .cm-link, .cm-s-inner.cm-link { color: rgb(0, 0, 204); }
.cm-negative { color: rgb(221, 68, 68); }
.cm-positive { color: rgb(34, 153, 34); }
.cm-header, .cm-strong { font-weight: 700; }
.cm-del { text-decoration: line-through; }
.cm-em { font-style: italic; }
.cm-link { text-decoration: underline; }
.cm-error { color: red; }
.cm-invalidchar { color: red; }
.cm-constant { color: rgb(38, 139, 210); }
.cm-defined { color: rgb(181, 137, 0); }
div.CodeMirror span.CodeMirror-matchingbracket { color: rgb(0, 255, 0); }
div.CodeMirror span.CodeMirror-nonmatchingbracket { color: rgb(255, 34, 34); }
.cm-s-inner .CodeMirror-activeline-background { background: inherit; }
.CodeMirror { position: relative; overflow: hidden; }
.CodeMirror-scroll { height: 100%; outline: 0px; position: relative; box-sizing: content-box; background: inherit; }
.CodeMirror-sizer { position: relative; }
.CodeMirror-gutter-filler, .CodeMirror-hscrollbar, .CodeMirror-scrollbar-filler, .CodeMirror-vscrollbar { position: absolute; z-index: 6; display: none; }
.CodeMirror-vscrollbar { right: 0px; top: 0px; overflow: hidden; }
.CodeMirror-hscrollbar { bottom: 0px; left: 0px; overflow: hidden; }
.CodeMirror-scrollbar-filler { right: 0px; bottom: 0px; }
.CodeMirror-gutter-filler { left: 0px; bottom: 0px; }
.CodeMirror-gutters { position: absolute; left: 0px; top: 0px; padding-bottom: 30px; z-index: 3; }
.CodeMirror-gutter { white-space: normal; height: 100%; box-sizing: content-box; padding-bottom: 30px; margin-bottom: -32px; display: inline-block; }
.CodeMirror-gutter-wrapper { position: absolute; z-index: 4; background: 0px 0px !important; border: none !important; }
.CodeMirror-gutter-background { position: absolute; top: 0px; bottom: 0px; z-index: 4; }
.CodeMirror-gutter-elt { position: absolute; cursor: default; z-index: 4; }
.CodeMirror-lines { cursor: text; }
.CodeMirror pre { border-radius: 0px; border-width: 0px; background: 0px 0px; font-family: inherit; font-size: inherit; margin: 0px; white-space: pre; overflow-wrap: normal; color: inherit; z-index: 2; position: relative; overflow: visible; }
.CodeMirror-wrap pre { overflow-wrap: break-word; white-space: pre-wrap; word-break: normal; }
.CodeMirror-code pre { border-right: 30px solid transparent; width: fit-content; }
.CodeMirror-wrap .CodeMirror-code pre { border-right: none; width: auto; }
.CodeMirror-linebackground { position: absolute; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; }
.CodeMirror-linewidget { position: relative; z-index: 2; overflow: auto; }
.CodeMirror-wrap .CodeMirror-scroll { overflow-x: hidden; }
.CodeMirror-measure { position: absolute; width: 100%; height: 0px; overflow: hidden; visibility: hidden; }
.CodeMirror-measure pre { position: static; }
.CodeMirror div.CodeMirror-cursor { position: absolute; visibility: hidden; border-right: none; width: 0px; }
.CodeMirror div.CodeMirror-cursor { visibility: hidden; }
.CodeMirror-focused div.CodeMirror-cursor { visibility: inherit; }
.cm-searching { background: rgba(255, 255, 0, 0.4); }
@media print {
  .CodeMirror div.CodeMirror-cursor { visibility: hidden; }
}


:root {
    --side-bar-bg-color: #fafafa;
    --control-text-color: #777;
}

@include-when-export url(https://fonts.loli.net/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext);

html {
    font-size: 16px;
}

body {
    font-family: "Open Sans","Clear Sans","Helvetica Neue",Helvetica,Arial,sans-serif;
    color: rgb(51, 51, 51);
    line-height: 1.6;
}

#write {
    max-width: 860px;
  	margin: 0 auto;
  	padding: 30px;
    padding-bottom: 100px;
}

@media only screen and (min-width: 1400px) {
	#write {
		max-width: 1024px;
	}
}

@media only screen and (min-width: 1800px) {
	#write {
		max-width: 1200px;
	}
}

#write > ul:first-child,
#write > ol:first-child{
    margin-top: 30px;
}

a {
    color: #4183C4;
}
h1,
h2,
h3,
h4,
h5,
h6 {
    position: relative;
    margin-top: 1rem;
    margin-bottom: 1rem;
    font-weight: bold;
    line-height: 1.4;
    cursor: text;
}
h1:hover a.anchor,
h2:hover a.anchor,
h3:hover a.anchor,
h4:hover a.anchor,
h5:hover a.anchor,
h6:hover a.anchor {
    text-decoration: none;
}
h1 tt,
h1 code {
    font-size: inherit;
}
h2 tt,
h2 code {
    font-size: inherit;
}
h3 tt,
h3 code {
    font-size: inherit;
}
h4 tt,
h4 code {
    font-size: inherit;
}
h5 tt,
h5 code {
    font-size: inherit;
}
h6 tt,
h6 code {
    font-size: inherit;
}
h1 {
    padding-bottom: .3em;
    font-size: 2.25em;
    line-height: 1.2;
    border-bottom: 1px solid #eee;
}
h2 {
   padding-bottom: .3em;
    font-size: 1.75em;
    line-height: 1.225;
    border-bottom: 1px solid #eee;
}
h3 {
    font-size: 1.5em;
    line-height: 1.43;
}
h4 {
    font-size: 1.25em;
}
h5 {
    font-size: 1em;
}
h6 {
   font-size: 1em;
    color: #777;
}
p,
blockquote,
ul,
ol,
dl,
table{
    margin: 0.8em 0;
}
li>ol,
li>ul {
    margin: 0 0;
}
hr {
    height: 2px;
    padding: 0;
    margin: 16px 0;
    background-color: #e7e7e7;
    border: 0 none;
    overflow: hidden;
    box-sizing: content-box;
}

li p.first {
    display: inline-block;
}
ul,
ol {
    padding-left: 30px;
}
ul:first-child,
ol:first-child {
    margin-top: 0;
}
ul:last-child,
ol:last-child {
    margin-bottom: 0;
}
blockquote {
    border-left: 4px solid #dfe2e5;
    padding: 0 15px;
    color: #777777;
}
blockquote blockquote {
    padding-right: 0;
}
table {
    padding: 0;
    word-break: initial;
}
table tr {
    border-top: 1px solid #dfe2e5;
    margin: 0;
    padding: 0;
}
table tr:nth-child(2n),
thead {
    background-color: #f8f8f8;
}
table tr th {
    font-weight: bold;
    border: 1px solid #dfe2e5;
    border-bottom: 0;
    margin: 0;
    padding: 6px 13px;
}
table tr td {
    border: 1px solid #dfe2e5;
    margin: 0;
    padding: 6px 13px;
}
table tr th:first-child,
table tr td:first-child {
    margin-top: 0;
}
table tr th:last-child,
table tr td:last-child {
    margin-bottom: 0;
}

.CodeMirror-lines {
    padding-left: 4px;
}

.code-tooltip {
    box-shadow: 0 1px 1px 0 rgba(0,28,36,.3);
    border-top: 1px solid #eef2f2;
}

.md-fences,
code,
tt {
    border: 1px solid #e7eaed;
    background-color: #f8f8f8;
    border-radius: 3px;
    padding: 0;
    padding: 2px 4px 0px 4px;
    font-size: 0.9em;
}

code {
    background-color: #f3f4f4;
    padding: 0 2px 0 2px;
}

.md-fences {
    margin-bottom: 15px;
    margin-top: 15px;
    padding-top: 8px;
    padding-bottom: 6px;
}


.md-task-list-item > input {
  margin-left: -1.3em;
}

@media print {
    html {
        font-size: 13px;
    }
    table,
    pre {
        page-break-inside: avoid;
    }
    pre {
        word-wrap: break-word;
    }
}

.md-fences {
	background-color: #f8f8f8;
}
#write pre.md-meta-block {
	padding: 1rem;
    font-size: 85%;
    line-height: 1.45;
    background-color: #f7f7f7;
    border: 0;
    border-radius: 3px;
    color: #777777;
    margin-top: 0 !important;
}

.mathjax-block>.code-tooltip {
	bottom: .375rem;
}

.md-mathjax-midline {
    background: #fafafa;
}

#write>h3.md-focus:before{
	left: -1.5625rem;
	top: .375rem;
}
#write>h4.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
#write>h5.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
#write>h6.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
.md-image>.md-meta {
    /*border: 1px solid #ddd;*/
    border-radius: 3px;
    padding: 2px 0px 0px 4px;
    font-size: 0.9em;
    color: inherit;
}

.md-tag {
    color: #a7a7a7;
    opacity: 1;
}

.md-toc { 
    margin-top:20px;
    padding-bottom:20px;
}

.sidebar-tabs {
    border-bottom: none;
}

#typora-quick-open {
    border: 1px solid #ddd;
    background-color: #f8f8f8;
}

#typora-quick-open-item {
    background-color: #FAFAFA;
    border-color: #FEFEFE #e5e5e5 #e5e5e5 #eee;
    border-style: solid;
    border-width: 1px;
}

/** focus mode */
.on-focus-mode blockquote {
    border-left-color: rgba(85, 85, 85, 0.12);
}

header, .context-menu, .megamenu-content, footer{
    font-family: "Segoe UI", "Arial", sans-serif;
}

.file-node-content:hover .file-node-icon,
.file-node-content:hover .file-node-open-state{
    visibility: visible;
}

.mac-seamless-mode #typora-sidebar {
    background-color: #fafafa;
    background-color: var(--side-bar-bg-color);
}

.md-lang {
    color: #b4654d;
}

.html-for-mac .context-menu {
    --item-hover-bg-color: #E6F0FE;
}

#md-notification .btn {
    border: 0;
}

.dropdown-menu .divider {
    border-color: #e5e5e5;
}

.ty-preferences .window-content {
    background-color: #fafafa;
}

.ty-preferences .nav-group-item.active {
    color: white;
    background: #999;
}

 .typora-export li, .typora-export p, .typora-export,  .footnote-line {white-space: normal;} 
</style>
    <!-- Bootstrap Layout css -->
    <link rel="stylesheet"; href="../bootstrap/3.4.1/css/bootstrap.min.css">
    
    <!-- Bootstrap Layout js -->
    <script src="../libs/jquery/3.4.1/jquery.min.js"></script>
    <script src="../bootstrap/3.4.1/js/bootstrap.min.js"></script>
</head>
<body class='typora-export os-windows' >
<div  id='write'  class = 'is-node'>
    
    <h1><a name="single-trial-parameter-fits" class="md-header-anchor"></a><span>Single-trial Parameter Fits</span></h1><p><span>This <a href="https://github.com/m053m716/rc-proj/tree/master/spikeAnalyses/%2Banalyze/%2Bstat"; style="font-weight:bold">package</a> contains </span><code>MATLAB</code><span> functions for two purposes:</span></p><ol start='' ><li><span>Fitting informative parameters to reduce time-series to two components</span></li><li><span>Exporting the fit components along with fit information and relevant grouping metadata in order to model the fit parameters, highlighting group differences during recovery that occur on the scale of single-channel multi-unit activity.</span></li></ol><h2><a name="overview" class="md-header-anchor"></a>
    
    <span>Overview</span></h2><p><span>This section indicates the fit parameters and generic database structure.</span></p><h3><a name="motivation" class="md-header-anchor"></a>
    
    <span>Motivation</span></h3><p><span>The main </span><strong><span>fixed effect</span></strong><span> of interest is the interaction of </span><em><span>Group, Electrode Location, and Post-Operative Day</span></em><span> in predicting changes in the following response parameters:</span></p><ul><li><p><strong><span>Time of spike rate peak</span></strong><span> (&quot;timing parameter&quot;)</span></p><ul><li><strong><span>Null hypothesis H0:</span></strong><span> No effect of post-operative day on timing parameter for either area.</span></li><li><strong><span>Alternative hypothesis H1:</span></strong><span> Timing parameter in premotor homolog has significant positive coefficient for interaction effect, corresponding to a &quot;shift&quot; in distribution of observed activation tunings as existing corticospinal projections become more active in direct-alignment to behavior as opposed to predominantly during behavior preparation. No effect on timing parameter in motor cortex homolog in uninjured hemisphere.</span></li><li><strong><span>Alternative hypothesis H2:</span></strong><span> Timing parameter in motor cortex homolog in uninjured hemisphere has significant </span><em><span>negative</span></em><span> coefficient for interaction effect, suggesting that descending corticospinal projections from the uninjured hemisphere become more active in modulating the behavior as recovery progresses.</span></li><li><strong><span>Alternative hypothesis H3:</span></strong><span>  Timing parameter in motor cortex homolog in uninjured hemisphere has significant </span><em><span>positive</span></em><span> coefficient for interaction effect, which could relate to an increased rate of successful pellet retrieval, requiring the use of the opposite forelimb as the pellet is moved towards the mouth.</span></li><li><strong><span>Alternative hypothesis H4:</span></strong><span> Some combination of </span><strong><span>H1</span></strong><span> and either </span><strong><span>H2</span></strong><span> or </span><strong><span>H3</span></strong><span>.</span></li></ul></li><li><p><strong><span>Frequency content of spike rate oscillation</span></strong><span> (&quot;frequency parameter&quot;)</span></p></li></ul>
    
    <h3><a name="database-structure" class="md-header-anchor"></a><span>Database Structure</span></h3><p><span>These data are structured as summarized in the </span><strong><span>Data</span></strong><span> tab as described in Figures 1-3 </span><strong><a href='https://m053m716.github.io/RC-Data/'><span>here</span></a></strong><span>. Specifically, data in </span><strong><span>Figure 3</span></strong><span> represent filtered views of rows from the </span><strong><span>main database table</span></strong><span>, </span><code>T</code><span>, wherein each row corresponds to the spike rate time-series estimate for a single recording channel, in alignment to a single behavioral event, from a single animal during one recording. In </span><code>MATLAB</code><span>, </span><code>T</code><span> contains the following variables (columns, listing most-relevant):</span></p>
    
    <ul><li><p><strong><code>RowID</code></strong><span> : A unique alphanumeric identifier for each row, used in table filtering operations to ensure that the correct metadata is retained with each row.</span></p></li><li><p><strong><code>Group</code></strong><span> : A categorical variable with 2 levels: {&quot;Ischemia&quot; or &quot;Intact&quot;}. This is one of the main effects we are interested in.</span></p></li><li><p><strong><code>AnimalID</code></strong><span> : Categorical variable that is nested within </span><strong><code>Group</code></strong><span>. By group, there are 4 rats in &quot;Intact&quot; and 6 rats in &quot;Ischemia&quot;.</span></p></li><li><p><strong><code>BlockID</code></strong><span> : Categorical variable that is nested within </span><strong><code>AnimalID</code></strong><span>. Each element denotes an individual recording session; each session was performed on a unique </span><strong><code>PostOpDay</code></strong><span>, which takes a value on the range [3,30]. </span><strong><code>BlockID</code></strong><span> is unique to each rat, but </span><strong><code>PostOpDay</code></strong><span> is not.</span></p></li><li><p><strong><code>Outcome</code></strong><span> : Categorical variable with 2 levels: {&quot;Successful&quot; or &quot;Unsuccessful&quot;}. We are restricting analyses to only &quot;Successful&quot; outcomes, since those present the most stereotyped behavior and we wish to compare changes in neurophysiology that facilitate a conserved behavior as recovery progresses.</span></p></li><li><p><strong><code>Trial_ID</code></strong><span> : Categorical variable indicating unique trials (from all recordings/animals). </span></p><ul><li><p><em><span>Note:</span></em><span> The following variables are &quot;blocked&quot; or nested within </span><strong><span>Trial_ID</span></strong><span>:</span></p><ul><li><strong><code>Alignment</code></strong><span> : Categorical event variable {&quot;Reach&quot;,&quot;Grasp&quot;,&quot;Support&quot;,&quot;Complete&quot;}, although we are mainly restricting the dataset to look at &quot;Reach&quot; and &quot;Grasp&quot; alignments.</span></li><li><strong><code>ChannelID</code></strong><span> : Categorical variable indicating unique identifier for a particular spatial sensor position within a single rat. Each rat contains 32 channels divided equally between hemispheres, although this number is reduced in order to only retain channels with neurophysiological activity observed during each recording session.</span></li></ul></li></ul></li><li><p><strong><code>ML</code></strong><span> : Categorical variable indicating whether shank was on &quot;Medial&quot; or &quot;Lateral&quot; row of probe array. Each member of </span><strong><code>ProbeID</code></strong><span> contains a unique subset of up to 16 elements of </span><strong><code>ChannelID</code></strong><span>. They are spatially configured in two rows, such that 8 of those elements are situated more medially relative to bregma (skull landmark used for stereotaxic positions), while 8 are more lateral. Note that </span><strong><code>ML</code></strong><span> should be the same for each instance of </span><strong><code>ChannelID</code></strong><span> (the same is true for any other &quot;channel-related&quot; metadata). The more-specific (continuous) location data for each </span><strong><code>ChannelID</code></strong><span> is shown in the variables </span><strong><code>X</code></strong><span> (anteroposterior stereotaxic distance from bregma, mm), and </span><strong><code>Y</code></strong><span> (mediolateral stereotaxic distance from bregma, mm).</span></p></li><li><p><strong><code>ICMS</code></strong><span> : The evoked movement response obtained by using an intracortical microstimulation test pulse prior to insertion of the electrode array, nearest to the co-registered insertion sites (Categorical: {&quot;DF&quot;, &quot;DF-PF&quot;, &quot;PF&quot;, &quot;O&quot;, &quot;NR&quot;} which correspond to {&quot;Distal Forelimb&quot;, &quot;Distal or Proximal Forelimb&quot;, &quot;Proximal Forelimb&quot;, &quot;Other&quot;, &quot;Non-responsive&quot;}).</span></p></li><li><p><strong><code>Area</code></strong><span> : The main effect of interest in combination with </span><strong><code>PostOpDay</code></strong><span> and </span><strong><code>Group</code></strong><span>; this has two levels {&quot;RFA&quot; (rostral forelimb area; PM homolog) or &quot;CFA&quot; (caudal forelimb area; M1 homolog)}. </span></p></li></ul>
    
    <h2><a name="parameter-fitting-procedure" class="md-header-anchor"></a><span>Parameter Fitting Procedure</span></h2>
    
    <h3><a name="syntax" class="md-header-anchor"></a><span>Syntax</span></h3><p><span>After recovering the master database, </span><code>T</code><span>, we recover the parameter fits using:</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="(matlab)"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="(matlab)"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">G = analyze.stat.get_fitted_table(T,min_n_trials);</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre>
    
    <p><span>Where </span><code>min_n_trials</code><span> is an adjustable parameter that has been set to </span><strong><span>5</span></strong><span> for the exported table, meaning that there must be at least 5 (successful) trials from a given element of </span><strong><code>BlockID</code></strong><span> in order for that recording to be considered for further analysis. This is to avoid cases where we have a single trial for a given channel on some post-operative day, which could really bias the data if we have spurious fluctuation or a noisy channel for whatever reason on that particular instance. Ideally the restriction number could be higher, but given the nature of these data (the rehabilitation testing is restricted to a limited number of attempts to begin with),  it&#39;s a compromise between causing spurious model associations and throwing away too much of the experiment.</span></p>
    
    <h3><a name="description" class="md-header-anchor"></a><span>Description</span></h3><p><span>The bulk of the </span><code>get_fitted_table</code><span> function is in the parameterized built-in </span><code>fmincon</code><span> function provided by the </span><code>Statistics and Machine Learning</code><span> toolkit. The algorithm is as follows:</span></p><ol start='' ><li><span>Set a &quot;hyperparameter&quot; grid, sweeping a combination of relevant parameters.</span></li><li><span>Evaluate at each point on the grid.</span></li><li><span>Initialize the constrained optimizer with the &quot;best&quot; combination (in this case, determined by the minimum residual sum of squares). </span></li></ol><p><span>The parameters are used by the </span><code>gauspuls</code><span> function (</span><code>Signal Processing Toolbox</code><span>) to fit a Gaussian-modulated cosine to the observed (smoothed) spike rate of each individual trial (data table row). Prior to the </span><code>fmincon</code><span> procedure, the individual trial spike rate is detrended using the following algorithm:</span></p><ol start='' ><li><p><span>Compute the trial median spike rate (</span><code>b</code><span>)</span></p></li><li><p><span>Recover the least-squares optimal coefficient (</span><code>a</code><span>) of time (</span><code>t</code><span>) as a linear predictor of spike rate (</span><code>Rate</code><span>) as </span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="(matlab)"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="(matlab)"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">a = (Rate - b)/t; % matlab performs least-squares regression</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre></li><li><p><span>The de-trended rates are therefore the residual rate not explained by the model</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="(matlab)"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="(matlab)"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">rate = a*t + b;</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre></li></ol>
    
    <p><span>Often, spike rate histograms in alignment to a motor event look like a series of peaks and valleys, indicating alternating periods where a spike is more likely (due to excitation related somehow to the behavior) or less likely (due to inhibition related to the behavior, or refractory period of having been recently likely to depolarize). This &quot;shape&quot; is captured well by </span><code>gauspuls</code><span>, which allows us to recover the following parameters: </span></p>
    
    <h4><a name="tau" class="md-header-anchor"></a><strong><span>tau</span></strong></h4>
    <p><span>An offset parameter, which describes the time-shift of a zero-centered Gaussian-modulated cosine estimated at the relative times for which the spike rate was obtained.</span></p>
    
    <h4><a name="omega" class="md-header-anchor"></a><strong><span>omega</span></strong></h4>
    <p><span>Frequency of the cosine component, which ranges on the order of a physiologically plausible behavior.</span></p><ul><li><span>For example, it&#39;s only reasonable that a rat might be expected to complete a reaching behavior at a maximum of 2 attempts per second. Therefore most values here are in the range of 0.25 - 1.5 Hz.</span></li></ul>
    
    <h4><a name="sigma" class="md-header-anchor"></a><strong><span>sigma</span></strong></h4>
    <p><span>Fractional bandwidth of </span><strong><a href='#omega'><span>omega</span></a></strong><span> for the Gaussian pulse, which describes the bandwidth at half-maximum amplitude of the Gaussian &quot;envelope.&quot;</span></p>
    
    <h3><a name="fit-figures"; class="md-header-anchor"; id="figures"></a>Figures</h3>
    <!-- Fit Examples: Tabbed -->
    <div class="row"; style="width:100%;height:650px">
        <div class="col-sm-3">
            <ul class="nav nav-pills nav-stacked">
                <li class="active"><a data-toggle="tab"; href="#fig1_tab"; id="#fig1_btn">Fig. 1: Example excluded fit</a></li>
                <li><a data-toggle="tab"; href="#fig2_tab"; id="#fig2_btn">Fig. 2: Error Distribution</a></li>
                <li><a data-toggle="tab"; href="#fig3_tab"; id="#fig3_btn">Fig. 3: Example good fit</a></li>
                <li><a data-toggle="tab"; href="#fig4_tab"; id="#fig4_btn">Fig. 4: Example moderate fit</a></li>
                <li><a data-toggle="tab"; href="#fig5_tab"; id="#fig5_btn">Fig. 5: Example poor fit</a></li>
                <li><a data-toggle="tab"; href="#fig6_tab"; id="#fig6_btn">Fig. 6: PCA summary of estimated parameters</a></li>
            </ul> 
        </div>            
        <div class="col-sm-9">
            <div class="tab-content"; style="width:100%;height:100%">
                <!--Figure 1 sub-pane -->
                <div class="tab-pane fade in active"; id="fig1_tab"; style="width:100%;height:100%">
                    <p><img src="https://media.githubusercontent.com/media/m053m716/RC-Data/master/Figures/img/FMINCON_Example_High_Bandwidth__Terrible_Fit.png"; referrerpolicy="no-referrer"; title="Figure 1: Example of poor fit with Envelope Bandwidth > 10-Hz"></p>
                    <p><strong><span>Figure 1: Example of poor Gaussian pulse fit.</span></strong></p>
                    <p>
                        One of the most-common parameter-based exclusions (due to lack of model fit) was when the Envelope Bandwidth converged to a value where the product of <strong><a href='#sigma'>sigma</a></strong> and <strong><a href='#omega'>omega</a></strong> exceeded 10-Hz, a value that is greater than the Nyquist Frequency given the bin width for generation of the initial spike rate histograms (60-ms). In this case, the fit (blue) captures a little bit of the modulation of the observed rate trajectory (black), but qualitatively doesn't capture the trajectory well, so such fits were excluded.
                    </p>
                </div>
                <!--Figure 2 sub-pane -->
                <div class="tab-pane fade"; id="fig2_tab"; style="width:100%;height:100%">
                    <p><img src="https://media.githubusercontent.com/media/m053m716/RC-Data/master/Figures/img/FMINCON_Error_Distribution.png"; referrerpolicy="no-referrer"; title="Figure 2: Distribution of Fit Errors"></p>
                    <p><strong><span>Figure 2: Distribution of fit errors.</span></strong></p>
                    <p>
                        After all <strong><a href='#exclusion_filters'>exclusion filters</a></strong> are applied to the fit parameters table, the errors are an approximately overdispersed Poisson distribution with mean of 2.1 and variance of 6.8.
                    </p>
                </div>
                <!--Figure 3 sub-pane -->
                <div class="tab-pane fade"; id="fig3_tab"; style="width:100%;height:100%">
                    <p><img src="https://media.githubusercontent.com/media/m053m716/RC-Data/master/Figures/img/FMINCON_Example_Good_Fit.png"; referrerpolicy="no-referrer"; title="Figure 3: Example of good model fit"></p>
                    <p><strong><span>Figure 3: Example of a trajectory that is well-fit by the Gaussian pulse model.</span></strong></p>
                    <p>
                        In this example, the sum of square errors is less than 1. The reconstructed trajectory (blue) capturing a majority of the observed data (black).
                    </p>
                </div>
                <!--Figure 4 sub-pane -->
                <div class="tab-pane fade"; id="fig4_tab"; style="width:100%;height:100%">
                    <p><img src="https://media.githubusercontent.com/media/m053m716/RC-Data/master/Figures/img/FMINCON_Example_Moderate_Fit.png"; referrerpolicy="no-referrer"; title="Figure 4: Example of good model fit"></p>
                    <p><strong><span>Figure 4: Example of a trajectory that isn't perfectly fit by the Gaussian pulse model, but still acceptable.</span></strong></p>
                    <p>
                        The reconstructed trajectory (blue) still captures the qualitative components of the observed data (black) that we were hoping to model, even with an error term that is approximately 3 times the mean error value.
                    </p>
                </div>
                <!--Figure 5 sub-pane -->
                <div class="tab-pane fade"; id="fig5_tab"; style="width:100%;height:100%">
                    <p><img src="https://media.githubusercontent.com/media/m053m716/RC-Data/master/Figures/img/FMINCON_Example_Poor_Fit.png"; referrerpolicy="no-referrer"; title="Figure 5: Example of poor model fit"></p>
                    <p><strong><span>Figure 5: Even relatively large error terms still capture the desired data aspects.</span></strong></p>
                    <p>
                        Toward the tail of the error distribution of the retained observations, the model (blue) still captures the data well (black).
                    </p>
                </div>
                <!--Figure 6 sub-pane -->
                <div class="tab-pane fade"; id="fig6_tab"; style="width:100%;height:100%">
                    <p><img src="https://media.githubusercontent.com/media/m053m716/RC-Data/master/Figures/img/FMINCON_PCA_Summary.png"; referrerpolicy="no-referrer"; title="Figure 6: Model Parameter PCA summary"></p>
                    <p><strong><span>Figure 6: PCA summary of fit model parameters.</span></strong></p>
                    <p>
                        <strong>(Top): </strong>Cumulative percent of data explained by each principal component (individual stem lines). The first three components capture approximately 90 percent of the parameter space.
                        <strong>(Bottom): </strong>Axes labels for this biplot correspond in color to the first three principal components from the top axes. 100 random observations are plotted as a scatter, showing that the majority of variance is explained by a component that is primarily weighted by the Peak Offset term <strong><a href='#tau'>tau</a></strong> while the third component may be most-heavily influenced by the frequency-related parameters (<strong><a href='#sigma'>sigma</a></strong> and <strong><a href='#omega'>omega</a></strong>). The constant and coefficient for linear de-trending of individual trial rate series are included in the PCA, here labeled as Premotor Bias and Linear Time Bias, respectively.
                    </p>
                </div>
            </div>
        </div>
    </div>
    
    <h3><a name="exported-table-statistics"; class="md-header-anchor"></a><span>Exported Table (statistics)</span></h3>
    <p><span>The table used to fit a </span><strong><span>Generalized Linear Mixed Effects model</span></strong><span> (Matlab function </span><code>fitglme</code><span>) is recovered from the table of estimated parameters using</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="(matlab)"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="(matlab)"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Gr = analyze.stat.remove_excluded(G);</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre>
    
    <h4><a name="parameter_orthogonality"; class="md-header-anchor"; id="parameter_pca"></a><strong><span>Orthogonality of Estimated Parameters</span></strong></h4>
    <p><span>This step initially performed a </span><strong><span>PCA</span></strong><span> to obtain components that qualitatively were orthogonal elements that when plotted using the built-in <strong><code><a href='#fig6_tab'; data-toggle="tab">biplot</a></code></strong> function to view the PC coefficients turned out to correspond to parameters that related to the timing and frequency parameters independently. Instead of fitting the principal component scores, since the timing and frequency parameter estimates do not demonstrate much covariance I decided to fit the recovered parameters directly.</span></p>
    
    <h4><a name="filtering_exclusions"; class="md-header-anchor"; id="exclusion_filters"></a><strong><span>Exclusion Filters</span></strong></h4>
    <p><span>The </span><code>remove_excluded</code><span> function also performs a series of filters that were included to reject poorly fit trajectories. Specifically, any time the product of </span><strong><span>sigma</span></strong><span> and </span><strong><span>omega</span></strong><span> is &gt; 10 Hz (which is above the Nyquist frequency given the spike rate bin size), the data is excluded. </span><code>Gr</code><span> only contains rows in alignment to </span><strong><span>Grasp</span></strong><span> events, where the pellet was </span><strong><span>Successfully</span></strong><span> retrieved. Visual inspection of fitted data indicates that the </span><code>gauspuls</code><span> fits are robust across the majority of the sum-of-square error range; however, there was an outlier cluster that had very bad fits and so any rows for which the error exceeded </span><strong><span>50</span></strong><span> (a.u.) the trials were excluded. Finally, any time that the peak offset parameter </span><strong><span>tau</span></strong><span> exceeded 750-ms or was lower than -1000-ms relative to the alignment event, the trial was excluded, as values outside that range become exceedingly likely to be associated with some other behavior that is not under consideration in this experiment. After all exclusions, the number of observations was reduced from 211,424 rows to 20,220 rows. Categorical exclusions (total: </span><strong><span>168,051</span></strong><span>) were performed first and breakdown (independently) as follows:</span></p><ul><li><em><span>By Outcome:</span></em><span> 117,611 exclusions</span></li><li><em><span>By Alignment:</span></em><span> 113,290 exclusions</span></li><li><em><span>By Day:</span></em><span> 0</span></li></ul><p><span>After categorical exclusions, removals from the remaining </span><strong><span>43,373</span></strong><span> observations due to parameter-fit rules (total: </span><strong><span>23,152</span></strong><span>) breakdown (independently) as follows:</span></p><ul><li><p><em><span>tau:</span></em><span> 21,396 exclusions</span></p><ul><li><em><span>tau &lt; 1 second:</span></em><span> 8,674 exclusions</span></li><li><em><span>tau &gt; 0.75 seconds:</span></em><span> 12,722 exclusions</span></li></ul></li><li><p><em><span>(sigma</span></em><span> * </span><em><span>omega) &gt; 10 Hz:</span></em><span>  2,617 exclusions</span></p></li><li><p><em><span>sum of square errors &gt; 50:</span></em><span> 0 exclusions</span></p></li></ul><p><span>The remaining </span><strong><span>20,221</span></strong><span> rows were included in the following table (</span><code>Gr</code><span> from the code above) for statistical analysis contains metadata described in </span><code>T</code><span> as well as the following </span><strong><span>endpoint (response)</span></strong><span> variables:</span></p><ul><li><strong><code>EnvelopeBW</code></strong><span> : Envelope bandwidth (Hz) - the product of </span><strong><span>sigma</span></strong><span> and </span><strong><span>omega</span></strong><span> parameters from model fit, which describes the bandwidth of overall modulation we see related to the behavior of interest.</span></li><li><strong><code>PeakOffset</code></strong><span> : Temporal offset (seconds) - The time of the fitted center of Gaussian pulse described by </span><strong><code>EnvelopeBW</code></strong><span>, relative to the behavioral alignment. Note that the cosine always has zero-phase angle at </span><code>t = PeakOffset</code><span> seconds relative to the event.</span></li></ul><p><span>In addition, the following covariates are included:</span></p><ul><li><p><strong><span>Error_SS</span></strong><span> : For each fit </span><code>rate</code><span> time-series, this is the sum of squares of errors at each time sample between the observed (smoothed) rate series and the fitted Gaussian-modulated cosine pulse. A higher value indicates a worse fit.</span></p></li><li><p><strong><span>Duration</span></strong><span> : Total duration of the behavior (time from </span><strong><span>Reach</span></strong><span> to </span><strong><span>Complete</span></strong><span> for a single trial; seconds). It has been well-described in the literature that there is a direct relationship between movement velocity and the population-level activity observed within the areas we are recording from. We expect this to be a strong covariate of the observed fit parameters, but we want to include it in the model to ensure that changes we are seeing are not simply due to the rat reaches becoming faster over time.</span></p><ul><li><em><span>Note: it is pretty clear that for the Ischemia group, the reaches </span><strong><span>do</span></strong><span> become faster over the time-course of the 3 weeks we do these recordings. So we definitely </span><strong><span>must</span></strong><span> account for this term.</span></em></li></ul></li></ul><p>&nbsp;</p></div>
</body>
</html>